### DISCUSSION TITLE
NextJS swcMinify is breaking arquero

### DISCUSSION THREAD

**Discussion by victorgdb** [uwdata/arquero#343] [General]
Hello, just to let you know that:

This code 
```
  let array = aq
    .from(data)
    .orderby(aq.desc('value'))
    .derive({
      cumulative: aq.rolling((d) => op.sum(d.value)),
    })
    .derive({
      share_cumulative: (d) => d.cumulative / op.sum(d.value),
    })
    .derive({
      category: (d) => (d.share_cumulative > 0.98 ? 'Other' : d.label),
    })
    .groupby('category')
    .rollup({
      value: (d) => op.sum(d.value),
    })
    .objects();
```
Produces the following error when minified
<img width="506" alt="f9f77596-15cf-4aaf-ba75-f89d0c3c52b0" src="https://github.com/uwdata/arquero/assets/437244/f22e2755-afec-46fb-8df6-2fa91c9ebf43">

To solve the issue I had to update next.config.js to:
```
const moduleExports = {
  reactStrictMode: true,
  swcMinify: false,
  experimental: {
    externalDir: true,
    typedRoutes: true,
  },
```
  
Or simply remove the faulty code.

Impacted versions:
```
    "arquero": "^5.2.0",
    "next": "^14.0.4",
```

**Comment by jheer** [comment]
Thanks! I converted this issue to a discussion post. Other tools may perform code transformations that are not compatible with Arquero. Thank you for documenting one potential fix!
